pkg/cli/config/config_test.go

```diff
@@ -59,3 +60,66 @@ func TestGetBackplaneConnection(t *testing.T) {
 		}
 	})
 }
+
+func Test_getConfigProxyList(t *testing.T) {
+
+	multiProxies := []interface{}{"-", "gellso", "https://secure-dummy-proxy.com", "http://squid.corp.redhat.com:"}
+
+	tests := []struct {
+		name           string
+		proxyURLConfig interface{}
+		want           []interface{}
+		wantErr        bool
+	}{
+		{"empty-proxy", "", []interface{}{""}, false},
+		{"single-proxy", "https://dummy-url", []interface{}{"https://dummy-url"}, false},
+		{"multi-proxy", multiProxies, multiProxies, false},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := getConfigProxyList(tt.proxyURLConfig)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("getConfigProxyList() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("getConfigProxyList() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestBackplaneConfiguration_getFirstWorkingProxyURL(t *testing.T) {
+
+	tests := []struct {
+		name    string
+		proxies []interface{}
+		want    string
+		wantErr bool
+	}{
+		{"invalid-format-proxy", []interface{}{""}, "", true},
+		{"multiple-invalid-proxies", []interface{}{"-", "gellso", ""}, "", true},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			svr := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				_, _ = w.Write([]byte("dummy data"))
+			}))
+
+			t.Setenv("BACKPLANE_URL", svr.URL)
+			t.Setenv("HTTPS_PROXY", "http://dummy.proxy")
```

Comment:
Is this set to test, that it's not defaulting to use the proxy environment variables?

Reply:
Rewrote the test